### PR TITLE
Switch docs theme to furo

### PR DESCRIPTION
**Pull request by marksweb**
## Description

Following discussion in the documentation work group, this switches the docs theme to furo.

### CODE REVIEW

**Review by crydotsnake — CHANGES_REQUESTED** [review]
We could change Divio AG too djangoCMS Association in the Copyright section

**Comment by vinitkumar** [comment]
@marksweb There are tests failing as well due to missing dependencies. Please fix them too.

**Comment by marksweb** [comment]
> @marksweb There are tests failing as well due to missing dependencies. Please fix them too.

That highlights a problem with those tests. The sphinx requirements are in the docs requirements, so the tests need reviewing that are failing as to what they're trying to do.

**Comment by marksweb** [comment]
@vinitkumar So it turns out there's a unit test which tests the docs, so there were some of the docs requirements in the base test requirements.

No need for this duplication, so I've removed the docs requirements from the test requirements and added an install step for the docs requirements in the unit tests. Things are starting to pass now. I'll address @crydotsnake point next

File: docs/conf.py [hunk 1]
```diff
@@ -64,7 +64,7 @@
 current_year = datetime.datetime.now().year
 # General information about the project.
 project = 'django cms'
-copyright = '2009-{}, Divio AG and contributors'.format(current_year)
+copyright = '2009-{}, djangoCMS Association and contributors'.format(current_year)
```
Comment (crydotsnake) [inline docs/conf.py:67]: Should we also change the year here? Because the djangoCMS Association exists since 2020 i think.

**Review by crydotsnake — CHANGES_REQUESTED** [review]

File: docs/conf.py (hunk 1 above)
Comment (FinalAngel) [reply docs/conf.py:67]: should be "django CMS Association" though

Comment (marksweb) [reply docs/conf.py]: @FinalAngel yeah good point, updated.

**Review by crydotsnake — APPROVED** [review]

**Review by vinitkumar — APPROVED** [review]
Looks good to me. Let's get this merged and released 🚢  🚀

**Pull request merged**